packages/material-ui-lab/src/SpeedDial/SpeedDial.js

```diff
@@ -3,14 +3,12 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import keycode from 'keycode';
 import warning from 'warning';
-import { withStyles } from '@material-ui/core/styles';
+import { duration, withStyles } from '@material-ui/core/styles';
 import Zoom from '@material-ui/core/Zoom';
-import { duration } from '@material-ui/core/styles/transitions';
 import Fab from '@material-ui/core/Fab';
-import { isMuiElement, setRef } from '@material-ui/core/utils/reactHelpers';
-import withForwardedRef from '@material-ui/core/utils/withForwardedRef';
+import { isMuiElement, setRef, withForwardedRef } from '@material-ui/core/utils';
 import * as utils from './utils';
-import clamp from '../utils/clamp';
+import { clamp } from '@material-ui/lab/utils';
```

Comment:
⚠️

Reply:
I guess the concern is this is different from how imports work currently. Should packages import themselves?

Reply:
Right. We can defer this to a later PR. By using actual paths we would make sure we're not importing from somewhere that wouldn't exist if downloaded. Could also be used to run test on the actual build. But that's for later. Will adjust.

Reply:
My concern is that we are importing lab from lab. Does that work?

Reply:
> My concern is that we are importing lab from lab. 

What is the concern with that?

Reply:
I doubt it will work with modern bundlers.

Reply:
This particular self import should work. The issue is that by allowing self imports it's pretty easy to introduce circular references.